chains/manager.go

```diff
@@ -393,39 +294,47 @@ func (m *manager) createChain(chainParams ChainParameters) {
 	// Note: Registering this after the chain has been tracked prevents a race
 	//       condition between the health check and adding the first chain to
 	//       the manager.
-	if chainParams.ID == constants.PlatformChainID {
+	if chain.Context.ChainID == constants.PlatformChainID {
 		if err := m.registerBootstrappedHealthChecks(); err != nil {
 			chain.Handler.StopWithError(context.TODO(), err)
 		}
 	}
 
 	// Tell the chain to start processing messages.
 	// If the X, P, or C Chain panics, do not attempt to recover
-	chain.Handler.Start(context.TODO(), !m.CriticalChains.Contains(chainParams.ID))
+	chain.Handler.Start(context.TODO(), !m.CriticalChains.Contains(chain.Context.XChainID))
 }
 
-// Create a chain
-func (m *manager) buildChain(chainParams ChainParameters, sb subnets.Subnet) (*chain, error) {
-	if chainParams.ID != constants.PlatformChainID && chainParams.VMID == constants.PlatformVMID {
+// New returns a new chain
+func (f *Factory) New(
+	chainID, subnetID, vmID ids.ID,
+	fxIDs []ids.ID,
+	genesis []byte,
+	beacons validators.Manager,
+	vmFactory vms.Factory,
+) (*Chain, error) {
```

Comment:
Should these factory functions be defined in the factory file? Or was the hope to minimize the diff in this PR?

Reply:
Yeah they should be. I did this to make the diff easier to review since it otherwise the diff becomes harder to review as you need to switch back and forth both files. I was planning on doing the move into the factory file in a follow-up once this was merged.